### ISSUE TITLE
Feature: Add new data source - harbor_projects

### ISSUE THREAD

**Issue by slushysnowman** [goharbor/terraform-provider-harbor#322]
**Is your feature request related to a problem? Please describe.**
We have a need to be able to read all Harbor projects through Terraform, so that we can assign permissions to relevant projects where we don't necessarily know the exact name of the project.

**Describe the solution you'd like**
A new data source `harbor_projects` with no required arguments, which will return the same output as `harbor_project`, but for all the projects in Harbor

**Comment by flbla** [comment]
Hi,
thanks for this feature request.

I can probably do something like this : 
```
data "harbor_projects" "name" {
  type = "ProxyCache" // optionnal (ProxyCache or Project)
  // name // optionnal regex
  // public // optionnal
  // vulnerability_scanning // optionnal
}
```

and return a list of projects

**Comment by flbla** [comment]
Hi @slushysnowman,
it will be available with release v3.10.0.
